dotnet/Miscreant/AesCtr.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Security.Cryptography;
+
+namespace Miscreant
+{
+	/// <summary>
+	/// Counter (CTR) mode, defined in NIST Special Publication
+	/// <see href="https://csrc.nist.gov/publications/detail/sp/800-38a/final">SP 800-38A</see>.
+	/// </summary>
+	public sealed class AesCtr : IDisposable
+	{
+		private const int BlockSize = Constants.BlockSize;
+		private const int KeyStreamBufferSize = 4096;
+
+		private readonly Aes aes;
+		private readonly ICryptoTransform encryptor;
+		private byte[] counter;
+		private ArraySegment<byte> keyStream;
+		private bool disposed;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="AesCtr"/> class with the specified key and initialization vector.
+		/// </summary>
+		/// <param name="key">The secret key for <see cref="AesCtr"> encryption.</param>
+		/// <param name="iv">The initialization vector for <see cref="AesCtr"> encryption.</param>
+		public AesCtr(byte[] key, byte[] iv)
+		{
+			if (key == null)
+			{
+				throw new ArgumentNullException(nameof(key));
+			}
+
+			if (iv == null)
+			{
+				throw new ArgumentNullException(nameof(iv));
+			}
+
+			if (iv.Length != BlockSize)
+			{
+				throw new CryptographicException("Specified initialization vector (IV) does not match the block size for this algorithm.");
+			}
+
+			aes = Aes.Create();
+			aes.Mode = CipherMode.ECB;
```

Comment:
It's unfortunate you have to implement it this way, but I double checked and .NET apparently does not provide an AES-CTR implementation?

Reply:
I wasn't happy about this either, but this was the only way - .NET really doesn't provide an AES-CTR implementation.